apps/ai/src/app/api/receive/route.ts

```diff
@@ -0,0 +1,149 @@
+import type { ParsedMail } from "mailparser";
+import { simpleParser } from "mailparser";
+import type { NextRequest } from "next/server";
+import { NextResponse } from "next/server";
+
+import prisma from "@calcom/prisma";
+
+import { env } from "../../../env.mjs";
+import { fetchAvailability } from "../../../tools/getAvailability";
+import { fetchEventTypes } from "../../../tools/getEventTypes";
+import { context } from "../../../utils/context";
+import getHostFromHeaders from "../../../utils/host";
+import now from "../../../utils/now";
+import sendEmail from "../../../utils/sendEmail";
+import { verifyParseKey } from "../../../utils/verifyParseKey";
+
+/**
+ * Verifies email signature and app authorization,
+ * then hands off to booking agent.
+ */
+export const POST = async (request: NextRequest) => {
+  const verified = verifyParseKey(request.url);
+
+  if (!verified) {
+    return new NextResponse("Unauthorized", { status: 401 });
+  }
+
+  const formData: any = await request.formData();
+  const body = Object.fromEntries(formData);
+
+  const signature = body.dkim;
```

Comment:
Can this be validated? Sendgrid offers an example payload of `"{@sendgrid.com : pass}"` - will this be empty on failure or maybe set to `fail`?

Reply:
Great catch - Will red team it with a non verified email and see what the payload is.

Reply:
Would hope that SendGrid doesn't forward emails that fail DKIM. But then the question is, what happens if DKIM is not configured? For security, DKIM setup and verification should be a hard requirement.

Reply:
Agreed.

See [dca8cd5](https://github.com/calcom/cal.com/pull/10992/commits/dca8cd54492b432ce15c9b7734b0fd9e0167ab54)